enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,560 @@
+    web server visible to the management controllers in the bare metal
+    hosts (because the management controllers are unlikely to be on
+    the same network as the external interfaces for the cluster
+    hosts). For vSphere or other hypervisors, that may be a filesystem
+    on the hypervisor host.
+11. The orchestration tool configures the host to boot the ISO. For
+    bare metal, that means configuring the baseboard management
+    controller (BMC) in each host to expose the ISO through the
+    virtual media interface. For vSphere or other hypervisors, similar
+    settings of the VM need to be adjusted.
+12. The orchestration tool boots the hosts.
+13. The orchestration tool uses `openshift-install agent wait-for` to
+    watch the deployment progress and wait for it to complete.
+14. On each host, the startup script in the image selects the correct
+    network settings based on the MAC addresses visible and applies
+    the network settings for each interface.
+15. On node0, the startup script in the image recognizes that it is on
+    node0 and launches the assisted service and the create-cluster
+    service to drive the deployment.
+16. On node0, the create-cluster service waits for the assisted
+    service to start, then creates an InfraEnv using the API.
+17. In parallel
+    - On node0, the create-cluster service waits for the number of
+      agents registered with the service to match the expected value
+      based on the input from the user.
+    - On all nodes (including node0), the startup script in the image
+      connects to the assisted service on node0 to find the InfraEnv
+      ID for the cluster being deployed (there will only be 1). After
+      it has the ID, the startup script launches the assisted agent
+      with that ID and configured to connect to the assisted service
+      on node0.
+18. All of the agents register their hosts with the assisted service.
+19. The assisted service performs its standard validations (network
+    connectivity, storage space, RAM, etc.) for all of the hosts.
+20. On node0, the create-cluster service sees that all known agents
+    have registered and the hosts have passed validation. It uses the
+    assisted service API to trigger the deployment to proceed.
+21. The assisted service performs the cluster deployment in its usual
+    way, except that the nodes will use the ISOs they booted from to
+    install RHCOS instead of fetching a new image from the assisted
+    service.
+22. The `openshift-install agent wait-for` process sees the OpenShift
+    API become available and starts using it to watch the cluster
+    complete deployment, combining the information provided by
+    `ClusterVersion` and `ClusterOperator` resources with the assisted
+    service progress API.
+23. On node0, when all of the other hosts have been deployed and
+    bootstrapping is complete, the agent reboots node0.
+24. The `openshift-install agent wait-for` process loses
```

Comment:
Where is the `openshift-install agent wait-for` process expected to run? If it was mentioned earlier in the document, I missed it.